### DISCUSSION TITLE
Lower level device management example

### DISCUSSION THREAD

**Discussion by rFronteddu** [freeconf/restconf#45] [Ideas]
I am trying to figure out how to have more control over the structure managed by Restconf. I am aware that I must define a better "manage" function but I am worried that the process will be time-consuming since I couldn't find clear examples of that.

For example, I could use an example of how I can lock updates and how I could save info inside a db instead of relying on a go structure.

I am also wondering if it wouldn't be easier to define the library in terms of callbacks and endpoints instead of feeding it a go struct and then managing the interaction with it. For example, I was imagining that if you have a field speed, you could have two callbacks one to get the value and one to update the value. Hopefully, I am not saying anything too silly :).

**Comment by rFronteddu** [comment]
I guess https://github.com/freeconf/examples/tree/master/node-extend could be the example I am looking for. We could add a readme to the examples project explaining what each example shows to make navigating easier.

**Comment by dhubler** [comment]
re: Example for building management code
I just pushed update to new reflection API
  https://freeconf.org/docs/examples/next-step/

re: Example README
This is pretty good legend to the examples
  https://freeconf.org/docs/examples/

re: callbacks, structs,....
I am not exactly following you, but it might be easier to build some code
and then point to a particular issue you are seeing.
